src/utils/MainApi.js

```diff
@@ -0,0 +1,78 @@
+export const BASE_URL = "http://localhost:3000";
+// export const BASE_URL =  "api.angelique.students.nomoredomainssbs.ru";
+
+function checkResponse(res) {
+  return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
+};
+
+export const register = (email, password, name) => {
+  return fetch(`${BASE_URL}/signup`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ email, password, name }),
+  })
+    .then((res) => checkResponse(res));
```

Comment:
**Might be better**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(checkResponse)
  }

```

And now you can safely replace `fetch` with `request` and remove repeated checking. And you don’t need to change anything else.